AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.kt

```diff
@@ -1308,7 +1309,10 @@ abstract class AbstractFlashcardViewer :
 
     private suspend fun automaticAnswerShouldWaitForAudio(): Boolean =
         withCol {
-            decks.configDictForDeckId(currentCard!!.did).waitForAudio
+            val card = currentCard ?: return@withCol false // Prevents null crash
+            val waitForAudio = decks.configDictForDeckId(card.did).waitForAudio
+            val soundConfig = CardSoundConfig.create(this, card) // Fetch autoplay setting
+            return@withCol waitForAudio && soundConfig.autoplay //  Only wait if autoplay is enabled
```

Comment:
the comments aren't necessary in this case
```suggestion
            val card = currentCard ?: return@withCol false
            val waitForAudio = decks.configDictForDeckId(card.did).waitForAudio
            val soundConfig = CardSoundConfig.create(this, card)
            return@withCol waitForAudio && soundConfig.autoplay
```